src/warp.jl

```diff
@@ -74,10 +74,13 @@ julia> indices(imgr)
 ```
 """
 function warp_new{T}(img::AbstractExtrapolation{T}, tform, inds::Tuple = autorange(img, inv(tform)))
-    out = OffsetArray(Array{T}(map(length, inds)), inds)
+    out = _allocate_array(T, inds)
     warp!(out, img, tform)
 end
 
+_allocate_array{T,N}(::Type{T}, inds::NTuple{N,Base.OneTo}) = Array{T}(map(length, inds))
+_allocate_array{T,N}(::Type{T}, inds::NTuple{N,AbstractUnitRange}) = OffsetArray(Array{T}(map(length, inds)), inds)
```

Comment:
The behavior of `collect` is changing in 0.6, see https://github.com/JuliaLang/julia/pull/21257. In particular we decided that `collect` was the right way to spell "throw away the custom indices and return an Array with the same values." (See https://github.com/JuliaArrays/IdentityRanges.jl/issues/1.)

Nevertheless I like this change overall. I would write the second, however, as `similar(Array{T}, inds)`. That way the indices are in charge of the output type. When `inds` are UnitRanges, then it will allocate an OffsetArray.